scipy/signal/spectral.py

```diff
@@ -2003,3 +2004,171 @@ def _median_bias(n):
     """
     ii_2 = 2 * np.arange(1., (n-1) // 2 + 1)
     return 1 + np.sum(1. / (ii_2 + 1) - 1. / ii_2)
+
+
+def hurst_dfa(x, s_min, s_max, plot=False, full=False):
+    """
+    Returns the Hurst exponent of a 1D array.
+
+    Parameters
+    ----------
+    x : array_like
+        Array or sequence containing the signal to be calculated.
+    s_min : int
+        Minimum window size. Must be positive.
+    s_max : int
+        Maximum window size. Cannot be larger than the length of x.
+    plot : bool, optional
+        Plot the graph from which the Hurst exponent is calculated.
+        Default is False.
+    full : bool, optional
+        Return the full fitting parameters describing the quality of the fit.
+
+    Returns
+    -------
+    hurst : float
+        The hurst exponent of the signal.
+    residuals, rank, singular_values, rcond
+        Only present if full=True. For more details, see np.linalg.lstsq
+
+    Raises
+    -------
+    ValueError
+        If s_min is negative or if s_max is longer than the input signal
+
+    See Also
+    -------
+    signal.waveforms.coloured_noise : Generate data with defined Hurst exponent
+    np.linalg.lstsq : Description of the "full" parameter
+
+    Examples
+    -------
+    Generate some pink noise, with alpha = 1 and H = (1+alpha)/2 = 1
+
+    >>> from scipy import signal
+    >>> pink_noise = signal.coloured_noise(beta='pink', size=100)
+
+    Calculate the Hurst exponent is correct
+
+    >>> hurst_exponent = hurst_dfa(pink_noise, s_min=5, s_max=50, plot=True)
+    1
+
+    Notes
+    -------
+    Noise and noisy signals can be quantified by their "colour", which has been
+    found to follow the form 1/f^a [1]_. The value of alpha quantifies the
```

Comment:
Consider rendering 1/f^a as math or code. Is a the same as alpha?

Reply:
Also I don't understand "found to follow". Does this mean that if I know f and I know a, I can calculate 1/f^a and this will represent a color? 
What is f? Consider defining f and alpha in the same sentence, e.g. "where the value of alpha quantifies...".

Reply:
I've rewritten this section a little bit to be clearer. f is the frequency of the signal, and alpha is a number that quantifies the colour of the signal. Hopefully this is clearer now - reading it back it was definitely unclear.